devicetypes/smartthings/zooz-multisiren.src/zooz-multisiren.groovy

```diff
@@ -0,0 +1,198 @@
+	}
+   
+	main "alarm"
+	details(["alarm", "humidity", "battery", "temperature", "tamper", "refresh"])
+	
+	}
+}
+
+def installed() {
+	initialize()
+}
+
+def refresh() {
+	def commands = []
+	 //get temperature value
+	commands << secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x01))
+	//get humidity value
+	commands << secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x05))
+	//get tamper state
+	commands << secure(zwave.notificationV3.notificationGet(notificationType: 0x07))
+	//get state of device (on or off)
+	commands << secure(zwave.basicV1.basicGet())
+	//get battery value
+	commands << secure(zwave.batteryV1.batteryGet())
+	
+	commands
+}
+
+def initialize() {
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: true, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	def cmd = []
+	//temperature and humidity are set for reporting every 60 min
+	cmd << secure(zwave.configurationV1.configurationSet(parameterNumber: 2, size: 2, configurationValue: [60]))
+	return response(cmd << refresh())
+}
+
+def configure() {
+	initialize()
+}
+
+def parse(String description) {
+	def result = null
+
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	createEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	createEvents(cmd)
+}
+
+private createEvents(cmd) {
+	[
+		createEvent([name: "switch", value: cmd.value ? "on" : "off"]),
+		createEvent([name: "alarm", value: cmd.value ? "both" : "off"])
+	]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	//when opening device cover
+	if(cmd.notificationType == 7) {
+		if(cmd.event == 3) {
+			sendEvent(name: "tamper", value: "detected")
+		} else {
+			sendEvent(name: "tamper", value: "clear")
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	if(cmd.sensorType == 1) {
+		sendEvent(name: "temperature", value: cmd.scaledSensorValue)
+	} else if(cmd.sensorType == 5) {
+		sendEvent(name: "humidity", value: cmd.scaledSensorValue)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	sendEvent(name: "battery", value: cmd.batteryLevel)
+}	
+	
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	[:]
+}
+
+def on() {
+	def commands = []
+	commands << secure(zwave.basicV1.basicSet(value: 0xFF))
+	commands << secure(zwave.basicV1.basicGet())
+	
+	delayBetween(commands, 100)
+}
+
+def off() {
+	def commands = []
+	commands << secure(zwave.basicV1.basicSet(value: 0x00))
+	commands << secure(zwave.basicV1.basicGet())
+	
+	delayBetween(commands, 100)
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	refresh()
```

Comment:
Since the purpose of this method is only to keep the device online and `refresh()` contains about 5 commands, for load reasons can we reduce the ping method to just a get of the switch/alarm state?

Reply:
sure :)